src/DynamoCore/Linting/LinterManager.cs

```diff
@@ -122,14 +134,18 @@ private void OnRuleEvaluated(IRuleEvaluationResult result)
                         if (storingResult.NodeIds != gRuleResult.NodeIds) 
                         {
                             // remove original result and replace with new one
-                            RuleEvaluationResults.Remove(storingResult);
-                            RuleEvaluationResults.Add(result);
+                            DynamoModel.OnRequestDispatcherInvoke(() =>
```

Comment:
what would ever call this not from the UI thread? And if it's not the UI thread, could it be multiple other threads -  do you need to add concurrency controls around your collection changes?

Reply:
The scenario we ran into so far was a rule that was watching `CachedValue` property change on a node.  This could be triggered from the SchedulerThread if a user had the graph in manual and pressed Run.

Reply:
At this time it does not seem like the concurrency guards are needed but @BogdanZavu and I still discussing that scenario

Reply:
Yes, at this point we only have main and scheduler thread involved and it's wither one or the other.
There is a chance in the future we might allow an api developer to have more control over events that trigger linter evaluation and at that point we will have to deal with concurrency.